### PR TITLE
Add populator for snapshot controller

### PR DESCRIPTION
**Pull request by NickrenREN**
Add populator for snapshot controller

### CODE REVIEW

**Comment by rootfs** [comment]
@tsmetana PTAL, thanks

File: pkg/controller/snapshotter/snapshotter.go [hunk 1]
```diff
@@ -39,14 +39,14 @@ const (
...
+// VolumeSnapshotter does the "heavy lifting": it spawns goroutines that talk to the
 // backend to actually perform the operations on the storage devices.
 // It creates and deletes the snapshots and promotes snapshots to volumes (PV). The create
 // and delete operations need to be idempotent and count with the fact the API object writes
 type VolumeSnapshotter interface {
-	CreateVolumeSnapshot(snapshotName string, snapshotSpec *tprv1.VolumeSnapshotSpec)
-	DeleteVolumeSnapshot(snapshotName string, snapshotSpec *tprv1.VolumeSnapshotSpec)
-	PromoteVolumeSnapshotToPV(snapshotName string, snapshotSpec *tprv1.VolumeSnapshotSpec)
+	CreateVolumeSnapshot(snapshotName string, snapshot *tprv1.VolumeSnapshot)
+	DeleteVolumeSnapshot(snapshotName string, snapshotc *tprv1.VolumeSnapshot)
```
Comment (tsmetana) [inline pkg/controller/snapshotter/snapshotter.go]: Small typo: "snapshotc". Moreover... Do we still need to pass the snaphsotName to all these functions? We have the whole VolumeSnapshot...

**Comment by NickrenREN** [comment]
@tsmetana  Thanks. I will send a new pr to try to simplify the interfaces.

**Comment by NickrenREN** [comment]
@tsmetana  typo fixed. And i will send a new PR to try to simplify the interfaces.

**Comment by rootfs** [comment]
thank you!

**Pull request merged**